### ISSUE TITLE
Feature: Add TIFF import / export support

### ISSUE THREAD

**Issue by geekrelief** [Seneral/TC2-Node-Painter-Beta#34]
Using https://bitmiracle.github.io/libtiff.net/ I hacked in TIFF import/export.  Unfortunately, I can't seem to edit TIFF properly in Affinity Photo if the MultiChannel Canvas uses the alpha channel.  Maybe the situation is different with Photoshop?  

There's the potential space savings possible when using zip compression in TIFF.

For a 2048x2048 file in RAW it's 32MB.  
![multichannel](https://cloud.githubusercontent.com/assets/27040/24122808/84873ad0-0d7a-11e7-913c-23e8fa2c03d1.png)

The TIFF is 16MB.
![file_size_comparison](https://cloud.githubusercontent.com/assets/27040/24122843/a5370184-0d7a-11e7-8367-41f65de405d0.png)

Attached code:
[Painting.zip](https://github.com/Seneral/TC2-Node-Painter-Beta/files/856445/Painting.zip)

**Comment by Seneral** [comment]
Cool, yes would love a 16Bit compressed format as export/import option...
Will take a look! :)

**Comment by geekrelief** [comment]
One down side is that Unity will attempt to import the tiff. :)  But, you could set the file extension to raw but still use TIFF. :)

**Comment by Seneral** [comment]
yes, something like that:)
Maybe not even tiff - any specific advantages this format has?

**Comment by geekrelief** [comment]
Well I was hoping to be able to export to TIFF for editing with an image editor.  I tried Affinity Photo and Krita and had issues with the alpha channel.  It was just an experiment.  But you're right it doesn't have to be TIFF.  I just happen to use it with MegaSplat for texturing plus there's that nice libTiff .net library.  It's about 700KB, so it made some sense.  

I tried looking at other formats, but I'm not sure if there's anything better.  Another option is just compressing the raw, too.

**Comment by Seneral** [comment]
Yes, makes sense - compressed raw can't be loaded into other programs though. Tiff looks promising in that regard, and that it's so small - but PNG or something similar would also be ok when it's small enough, right?

**Comment by geekrelief** [comment]
Does PNG handle 16-bit / 32-bit per channel?

**Comment by Seneral** [comment]
Yes, unity just doesn't import/export it as is

**Comment by Seneral** [comment]
Does the tiff library handle most major formats? Being only 700KB it doesn't seems so - but even if not, would still work for internal caching atleast:)
Would definitely give this a try - have the net port from Bitmiracle currently :https://bitmiracle.com/libtiff/

**Comment by Seneral** [comment]
Also, have to correct me on PNG - it supports 16Bit/channel max, which is enough.
Currently, Node Painter uses 16Bit internally but I plan to increase it again to 32Bit internally to decrease inaccuracies due to small paint values - but still exporting/saving as 16Bit to save memory (no sense in using 32Bit 99% of the time).

**Comment by geekrelief** [comment]
I'm no expert on TIFF. I was surprised how complicated it is compared to PNG!  But it seems to handle any kind of graphic data you'd want to manipulate.  So even if users can't edit the TIFF it's still a win in terms of space for a tradeoff in time.  Maybe I need a bigger harddrive?  :)

Of course you can can also use it for internal caching. :)

**Comment by Seneral** [comment]
Yes, I'm searching for some kind of internal compression I can use for 1. all entries of the undo system while they are in memory and 2. the cache and normal export files
It's ciritical that it's fast, most importantly, atleast for undo.

**Comment by geekrelief** [comment]
How does the undo system work?  Is it snapshotting?  Delta based?

**Comment by geekrelief** [comment]
Btw, not sure if you can use it but FLIF looks cool :) http://flif.info/index.html https://github.com/purenewman/FLIFSharp

**Comment by Seneral** [comment]
Indeed very interesting, but not much support and not optimized performance - author says, speed is ok but could be improved. Speed is critical for us though... still will take a look:)
